### ISSUE TITLE
AtomSpace benchmark crashes in python code

### ISSUE THREAD

**Issue by linas** [opencog/opencog#1245]
I get this crash:

```
opencog/benchmark/atomspace_bm -A
```
in the python code:
```
(gdb) bt
#0  0x0000000000000000 in ?? ()
#1  0x00007ffff70335f6 in opencog::PythonEval::getPyAtomspace (this=0x67b360, 
    atomspace=0x0)
    at /home/linas/src/novamente/src/opencog-git/opencog/cython/PythonEval.cc:142
#2  0x00007ffff70336a9 in opencog::PythonEval::init (this=0x67b360)
    at /home/linas/src/novamente/src/opencog-git/opencog/cython/PythonEval.cc:120
#3  0x00007ffff70339dc in opencog::PythonEval::eval_expr (this=0x67b360, 
    partial_expr=...)
    at /home/linas/src/novamente/src/opencog-git/opencog/cython/PythonEval.cc:422
#4  0x0000000000413782 in opencog::PythonEval::eval (this=0x67b360, expr=...)
    at /home/linas/src/novamente/src/opencog-git/opencog/cython/PythonEval.h:130
#5  0x000000000041248c in opencog::AtomSpaceBenchmark::startBenchmark (
    this=this@entry=0x7fffffffd720, numThreads=numThreads@entry=1)
    at /home/linas/src/novamente/src/opencog-git/opencog/benchmark/AtomSpaceBenchmark.cc:341
#6  0x000000000040c3be in main (argc=2, argv=0x7fffffffd918)
    at /home/linas/src/novamente/src/opencog-git/opencog/benchmark/atomspace_bm.cc:169
```

**Comment by inflector** [comment]
The PythonEval bug is fixed in: https://github.com/opencog/opencog/pull/1286

The benchmark still SEGFAULTs but only after running the Python benchmark and a few other benchmarks successfully. So since the new SEGFAULT now comes from the C++ AtomSpace code and appears not to be related to Python or Cython, I'm probably not the best one to fix it at the moment.

I suggest we close this issue after the Pull Request is accepted. I can then create a new and separate issue with the new stack trace.

**Comment by inflector** [comment]
Added new issue: https://github.com/opencog/opencog/issues/1289

**Comment by linas** [comment]
fixed
